processor/spanmetricsprocessor/processor_test.go

```diff
@@ -323,9 +364,29 @@ func newProcessorImp(mexp *mocks.MetricsExporter, tcon *mocks.TracesConsumer, de
 	}
 }
 
+// verifyConsumeMetricsInputCumulative expects one accumulation of metrics, and marked as cumulative
+func verifyConsumeMetricsInputCumulative(t testing.TB, input pdata.Metrics) bool {
+	return verifyConsumeMetricsInput(t, input, pdata.MetricAggregationTemporalityCumulative, 1)
+}
+
+// verifyConsumeMetricsInputDelta expects one accumulation of metrics, and marked as delta
+func verifyConsumeMetricsInputDelta(t testing.TB, input pdata.Metrics) bool {
+	return verifyConsumeMetricsInput(t, input, pdata.MetricAggregationTemporalityDelta, 1)
+}
+
+// verifyMultipleCumulativeConsumptions expects the amount of accumulations as kept track of by numCumulativeConsumptions.
+// numCumulativeConsumptions acts as a multiplier for the values, since the cumulative metrics are additive.
+func verifyMultipleCumulativeConsumptions() func(t testing.TB, input pdata.Metrics) bool {
+	numCumulativeConsumptions := 0
+	return func(t testing.TB, input pdata.Metrics) bool {
+		numCumulativeConsumptions++
+		return verifyConsumeMetricsInput(t, input, pdata.MetricAggregationTemporalityCumulative, numCumulativeConsumptions)
+	}
+}
+
 // verifyConsumeMetricsInput verifies the input of the ConsumeMetrics call from this processor.
 // This is the best point to verify the computed metrics from spans are as expected.
-func verifyConsumeMetricsInput(input pdata.Metrics, t *testing.T) bool {
+func verifyConsumeMetricsInput(t testing.TB, input pdata.Metrics, expectedTemporality pdata.MetricAggregationTemporality, numCumulativeConsumptions int) bool {
```

Comment:
For my learning, what is the benefit of using `testing.TB` over `testing.T`? I couldn't see benchmarking tests added in the PR.

Reply:
I advocated for to be used here so that it could be used in benchmarks in future :) 

I believe we have more work to add to this component so we wanted to make it as flexible as needed.